jenkins/coldlaunch.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash -x
+
+# receive list of apps to run coldlaunch test against
+# i.e. 'calendar clock dialer email'
+APP_LIST=$1
+
+post_to_treeherder() {
+  RAPTOR_BUILD_STATE=$1
+  RAPTOR_APP_NAME=$2
+  TEST_START_TIME=$3
+  TEST_FINISH_TIME=$4
+  TEST_FAILURE=$5
+
+  python --version
+  source raptor-env/bin/activate
+  python --version
+
+  cd post-to-treeherder
+  ./submit-to-treeherder.py \
+    --repository b2g-inbound \
+    --build-state ${RAPTOR_BUILD_STATE} \
+    --treeherder-url https://treeherder.allizom.org/ \
+    --treeherder-client-id raptor \
+    --treeherder-secret ${RAPTOR_TREEHERDER_SECRET} \
+    --test-type cold-launch \
+    --app-name ${RAPTOR_APP_NAME} \
+    --start-time ${TEST_START_TIME} \
+    --finish-time ${TEST_FINISH_TIME} \
+    --test-failure ${TEST_FAILURE}
+}
+
+run_coldlaunch_test() {
+  APP_ORIGIN=$1
+  ENTRY_POINT=$2
+
+  APP_FAILURE=0
+
+  FILE_FORMAT="${APP_ORIGIN}"
+  ENTRY_FLAG=""
+
+  # run coldlaunch
+  if [ -n "$ENTRY_POINT" ] ; then
+    FILE_FORMAT="${APP_ORIGIN}-${ENTRY_POINT}"
+    ENTRY_FLAG="--entry-point ${ENTRY_POINT}"
+  fi
+
+  if [ -n "$RAPTOR_DATABASE_NAME" ] ; then
+    DATABASE_FLAG="--database ${RAPTOR_DATABASE_NAME}"
+  fi
+
+  DEBUG=* node_modules/.bin/raptor test coldlaunch \
+    --serial ${ANDROID_SERIAL} \
+    --app ${APP_ORIGIN} \
+    --runs ${RUNS} \
+    --timeout ${TIMEOUT} \
+    --time ${TEST_TIME} \
+    --metrics ${FILE_FORMAT}.ldjson \
+    --logcat ${FILE_FORMAT}.logcat \
+    ${ENTRY_FLAG} 2>&1 | tee ${FILE_FORMAT}.log
+
+  LINES="$(wc -l < ${FILE_FORMAT}.ldjson)"
+  EXPECTED="$((RUNS + 1))"
+
+  # subit results to db
+  if [[ ${LINES} -eq ${EXPECTED} ]] ; then
+    DEBUG=* node_modules/.bin/raptor submit ${FILE_FORMAT}.ldjson \
+      --host ${RAPTOR_HOST} \
+      --port ${RAPTOR_PORT} \
+      --username ${RAPTOR_USERNAME} \
+      --password ${RAPTOR_PASSWORD} \
+      --protocol ${RAPTOR_PROTOCOL} \
+      ${DATABASE_FLAG}
+  else
+    APP_FAILURE=1
+    SUITE_FAILURE=1
+  fi
+}
+
+# main coldlaunch loop
+SUITE_FAILURE=0
+
+for i in ${APP_LIST[@]}
+do
+  APP=${i}
+
+  if [[ "${APP}" == "contacts" ]] || [[ "${APP}" == "dialer" ]] ; then
+    ORIGIN='communications'
+    ENTRY=$APP
+  else
+    ORIGIN=$APP
+    ENTRY=''
+  fi
+
+  APP_START_TIME=$(node -e "console.log(Date.now());")
+  post_to_treeherder running ${APP} ${APP_START_TIME}
+
+  run_coldlaunch_test ${ORIGIN} ${ENTRY}
+
+  APP_FINISH_TIME=$(node -e "console.log(Date.now());")
+  post_to_treeherder completed ${APP} ${APP_START_TIME} ${APP_FINISH_TIME} ${APP_FAILURE}
+done
+
+exit ${SUITE_FAILURE}
```

Comment:
Yessir! Only catch is, this script has to be the last one executed in the Jenkins build though, because if there's an exit 1 then I believe the jenkins build will stop here.